resources/js/components/BalanceList.vue

```diff
@@ -0,0 +1,110 @@
+<template>
+    <div class="container mx-auto py-12">
+        <Alert v-if="showAlert" :msg="`We're importing ${csvRowCount} balance entries. Sit tight.`" />
+
+        <template v-if="!modifying">
+            <Expense
+                v-for="[time, data] in Object.entries(expenseData.data)"
+                :key="data.id"
+                :data="data"
+                :datetime="time"
+            />
+        </template>
+
+        <Paginate :data="expenseData" />
+    </div>
+</template>
+
+<script>
+  import Expense from '../components/Expense.vue';
+  import Nav from '../components/Nav.vue';
+  import Summary from '../components/Summary.vue';
+  import Paginate from '../components/Paginate.vue';
+  import {
+    RESET_ENTRY,
+    DELETE_ENTRY,
+    RESET_TOTAL,
+    UPDATE_ENTRY,
+    SET_ROW_COUNT,
+    SET_IMPORT_ALERT
+  } from '../constants';
+  import { DateTime } from 'luxon';
+  import Alert from './Alert';
+
+  export default {
+    components: {
+      Alert,
+      Expense,
+      Paginate,
+      Summary,
+      Nav,
+    },
+
+    created() {
+      this.expenseData = JSON.parse(JSON.stringify(this.expenses));
+
+      this.$bus.$on(UPDATE_ENTRY, data => {
```

Comment:
These callbacks could probably be moved to component methods to clean things up a bit.